src/NumericalAlgorithms/RootFinding/GslMultiRootFunctionFdf.cpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/RootFinding/GslMultiRootFunctionFdf.hpp"
+
+template <size_t Dim>
+int Gsl_MultiRootFunctionFdf_Wrapper_f(const gsl_vector* v, void* params,
+                                       gsl_vector* func) noexcept {
```

Comment:
Mark the arguments const.  (The variables themselves, not the pointed-to data.)
`params` might be able to be `const void* const`, but void has weird conversion rules and I can't remember if that works.